dask-gateway-server/dask_gateway_server/backends/db_base.py

```diff
@@ -1443,7 +1443,9 @@ def get_scheduler_command(self, cluster):
         ]
 
     def worker_nthreads_memory_limit_args(self, cluster):
-        return str(cluster.config.worker_threads), str(cluster.config.worker_memory)
+        return str(cluster.config.worker_threads or cluster.config.worker_cores), str(
```

Comment:
This should be sufficient to handle the fallback to `worker_cores`. I need to double-check it's necessary (ideally the fallback would happen in the config class). I haven't used traitlets in a while though.

Reply:
I think these checks are actually necessary. I don't see an easy way to ensure that `config.worker_threads` is always not `None`, even if the user passes `worker_threads=None`. It seems like custom `_validate` checks aren't called when the value is None, so we would perhaps need a new trait type. Not worth it IMO.

Reply:
I don't like copying around this `or` bit everywhere, mind if I attempt to fix this at the traitlets level?

Reply:
Yeah, I'm happy to take care of it though (I wrote a bit of prototype code and threw it away)

Reply:
0d152f1c24747386a5bf87f85afa0a89f5b186a9 has the change if you have a chance to look. I wasn't wild this `WorkerThread` trait implicitly depending on the object having a `worker_cores` attribute to fall back to. Maybe traitlets has a better way to handle this?

Will push up a change removing these `or` bits later.

Reply:
I believe I have a fix for this (and a few other cleanups). Will push soon.